src/kubectlutils.ts

```diff
@@ -17,157 +17,123 @@
 import * as k8s from 'vscode-kubernetes-tools-api';
 import * as config from './config';
 
-export function isKubernetesAvailable(): Promise<boolean> {
-	return new Promise<boolean>( (resolve) => {
-		k8s.extension.kubectl.v1
-			.then( (kubectl) => {
-				resolve(kubectl && kubectl.available);
-			})
-			.catch( err => {
-				resolve(false);
-			});
-	});
+export async function isKubernetesAvailable(): Promise<boolean> {
+	const kubectl: k8s.API<k8s.KubectlV1> = await k8s.extension.kubectl.v1;
+	return kubectl && kubectl.available;
 }
 
-export function getNamedListFromKubernetes(itemType : string, extra? : string): Promise<string> {
-	return new Promise<string>( async (resolve, reject) => {
-		await k8s.extension.kubectl.v1
-			.then( async (kubectl) => {
-				let cmd = `get ${itemType}`;
-				if (extra) {
-					cmd += ` ${extra}`;
-				}
-				if (kubectl && kubectl.available) {
-					return await kubectl.api.invokeCommand(cmd);
-				} else {
-					reject(new Error('Kubernetes not available'));
-				}
-			})
-			.then( (result) => {
-				if (!result || result.code !== 0) {
-					let error = `Unable to invoke kubectl to retrieve ${itemType}`;
-					if (result && result.stderr) {
-						error = result.stderr;
-					}  
-					reject(error);
-				} else if (result) {
-					const splitResults = result.stdout;
-					resolve(splitResults);
-				}
-			})
-			.catch( (err) => reject(err) );
-	});
+export async function getNamedListFromKubernetes(itemType : string, extra? : string): Promise<string> {
+	const kubectl: k8s.API<k8s.KubectlV1> = await k8s.extension.kubectl.v1;
+	let cmd = `get ${itemType}`;
+	if (extra) {
+		cmd += ` ${extra}`;
+	}
+	if (kubectl && kubectl.available) {
+		const result: k8s.KubectlV1.ShellResult | undefined = await kubectl.api.invokeCommand(cmd);
+		if (!result || result.code !== 0) {
+			let error = `Unable to invoke kubectl to retrieve ${itemType}`;
+			if (result && result.stderr) {
+				error = result.stderr;
+			}  
+			return error;
```

Comment:
shouldn't we throw an error instead of returning it as a strign like if everything went fine?